enhancements/update/cluster-version-operator-x509-upstream-trust.md

```diff
@@ -0,0 +1,202 @@
+The fact that the current proxy configuration does not clearly address proxies defined at the network-routing level, where `httpsProxy` may be empty but additional trust still needs to be injected is out of scope.
+
+## Proposal
+
+A new config type `` will be added consuming the existing [`TLSSecurityProfile`][api-tls-security-profile] and [`ConfigMapNameReference`][api-config-map-name-reference].
+The `securityProfile` property and its comment are copied from [the existing `APIServerSpec` consumer][api-server-tls-security-profile] with the leading `TLS` dropped because `TLSConfig` is already TLS-specific.
+The `TrustedCA` property and its comment are based on [the existing `Proxy` consumer][api-proxy-trusted-ca].
+
+```go
+type TLSConfig {
+  // securityProfile specifies settings for TLS connections for externally exposed servers.
+  //
+  // If unset, a default (which may change between releases) is chosen. Note that only Old and
+  // Intermediate profiles are currently supported, and the maximum available MinTLSVersions
+  // is VersionTLS12.
+  // +optional
+  SecurityProfile *TLSSecurityProfile `json:"securityProfile,omitempty"`
+
+	// trustedCA is a reference to a ConfigMap containing a PEM-encoded X.509
+	// CA certificate bundle.  The consuming operator is responsible for:
+	//
+	// 1. Reading and validating the certificate bundle from the
+	//    required key "ca-bundle.crt".
+	// 2. Optionally merging it with the system default trust bundle, as
+	//    defined by the TLSConfig consumer.
+	// 3. Feeding the the resulting trust bundle to the appropriate
+	//    operand, as defined by the TLSConfig consumer.  This may involve
+	//    writing a ConfigMap in an operator-managed namespace that is then
+	//    volume-mounted into operand pods.
+	//
+	// The namespace for the ConfigMap referenced by trustedCA is
+	// "openshift-config". Here is an example ConfigMap (in yaml):
+	//
+	// apiVersion: v1
+	// kind: ConfigMap
+	// metadata:
+	//  name: user-ca-bundle
+	//  namespace: openshift-config
+	//  data:
+	//    ca-bundle.crt: |
+	//      -----BEGIN CERTIFICATE-----
+	//      Custom CA certificate bundle.
+	//      -----END CERTIFICATE-----
+	//
+	// +optional
+	TrustedCA *ConfigMapNameReference `json:"trustedCA,omitempty"`
+}
+```
+
+[`ClusterVersionSpec`][api-spec] will be extended to include the following new property:
+
+```go
+// upstreamTLS specifies settings for TLS connections to the upstream
+// update service.  If trustedCA is set, it is used directly and not
+// merged with the system default trust bundle.  If upstreamTLS is set,
+// even if none of its constituent properties are set, it takes
+// precedence over the proxy configuration.  To use the proxy
+// configuration for connections to the upstream update service, unset
+// upstreamTLS completely.
+// +optional
+UpstreamTLS *TLSConfig `json:"upstreamTLS,omitempty"
```

Comment:
I thought this was about trying to specify a way to connect to a custom cincinnati, in which case I expected to see a URL, a CA bundle, and perhaps clientcert/bearer token.

Reply:
> I thought this was about trying to specify a way to connect to a custom cincinnati, in which case I expected to see a URL...

This property already exists in `upstream`.

> ... a CA bundle...

I'm adding this with `upstreamTLS.securityProfile` in this enhancement.

> ... and perhaps clientcert/bearer token.

I'm punting on that for now; we can always add it to `TLSConfig` in a follow-up enhancement.

As of faa4a20, these points are all called out in the *Motivation* section where I point to either an existing property, a property added by this enhancement, or a property I punt as out of scope (for now).